specs/eip4844/validator.md

```diff
@@ -79,27 +78,30 @@ def validate_blobs_and_kzg_commitments(execution_payload: ExecutionPayload,
 
 3. If valid, set `block.body.blob_kzg_commitments = blob_kzg_commitments`.
 
-#### Constructing the `SignedBeaconBlockAndBlobsSidecar`
-To construct a `SignedBeaconBlockAndBlobsSidecar`, a `signed_beacon_block_and_blobs_sidecar` is defined with the necessary context for block and sidecar proposal.
-
-##### Block
-Set `signed_beacon_block_and_blobs_sidecar.beacon_block = block` where `block` is obtained above.
+#### Constructing the `SignedBlobSidecar`
+To construct a `SignedBlobSidecar`, a `signed_blob_sidecar` is defined with the necessary context for block and sidecar proposal.
 
 ##### Sidecar
-Coupled with block, the corresponding blobs are packaged into a sidecar object for distribution to the network.
 
-Set `signed_beacon_block_and_blobs_sidecar.blobs_sidecar = sidecar` where `sidecar` is obtained from:
+Coupled with block, the corresponding blobs are packaged into sidecar objects for distribution to the network.
+
+Each `sidecar` is obtained from:
 ```python
-def get_blobs_sidecar(block: BeaconBlock, blobs: Sequence[Blob]) -> BlobsSidecar:
-    return BlobsSidecar(
-        beacon_block_root=hash_tree_root(block),
-        beacon_block_slot=block.slot,
-        blobs=blobs,
-        kzg_aggregated_proof=compute_aggregate_kzg_proof(blobs),
-    )
+def get_blob_sidecar(block: BeaconBlock, blobs: Sequence[Blob]) -> Sequence[BlobsSidecar]:
+  return [
+    BlobsSidecar(
+      block_root=hash_tree_root(block),
+      index=idx
+      slot=block.slot,
+      block_parent_root=block.parent_root,
```

Comment:
If this is meant for shuffling derivation, why not set to `dependant_root`?

Reply:
There were points both in favor and against this:

* + better chance of having the shuffling
* - less information about the blob -> higher risk of spam
* - novel construct (`dependent_root` does not appear in any other p2p message)

In the end, I opted to replicate the block conditions as closely as possible for the blobs, so that blocks and blobs share as much "flow" as possible.

Reply:
I agree that parent_root feels simpler. And in most cases that should be available and you can derive the dependent root from it